### PR TITLE
MULE-11709: Remove *-processing-strategy elements

### PR DESCRIPTION
**Pull request by lucianoRM**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/construct/FlowProcessingStrategyTestCase.java [hunk 1]
```diff
@@ -44,6 +50,8 @@
 
   private Flow flow;
 
+
+
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/construct/FlowProcessingStrategyTestCase.java]: remove?

File: core-tests/src/test/java/org/mule/runtime/core/construct/FlowProcessingStrategyTestCase.java [hunk 2]
```diff
@@ -94,6 +102,37 @@ public void createDefaultProcessingStrategyIfNoneSpecified() throws Exception {
                is(instanceOf(new TransactionAwareWorkQueueProcessingStrategyFactory().getProcessingStrategyType())));
   }
 
+  @Test
+  public void processingStrategySetBySystemPropertyOverridesDefault() throws Exception {
+    testWithSystemProperty("org.mule.runtime.core.api.processor.strategy.ProcessingStrategyFactory",
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/construct/FlowProcessingStrategyTestCase.java]: use class.getName() ?

File: core-tests/src/test/java/org/mule/runtime/core/internal/construct/DefaultFlowBuilderTestCase.java [hunk 3]
```diff
@@ -59,12 +61,12 @@ public void setUp() throws Exception {
...
     when(muleConfiguration.getDefaultProcessingStrategyFactory()).thenReturn(defaultProcessingStrategyFactory);
     when(defaultProcessingStrategyFactory.create(any(), any())).thenReturn(processingStrategy);
+    when(muleContext.getSchedulerBaseConfig()).thenReturn(SchedulerConfig.config());
   }
 
   @Test
   public void buildsSimpleFlow() throws Exception {
-    Flow flow = flowBuilder.build();
-
+    Flow flow = flowBuilder.processingStrategyFactory(defaultProcessingStrategyFactory).build();
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/internal/construct/DefaultFlowBuilderTestCase.java]: Why is this needed? Flow behaviour hasn't changed has it?

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ApplicationModel.java [hunk 4]
```diff
@@ -120,8 +120,7 @@
   public static final String PROCESSOR_REFERENCE_ELEMENT = "processor";
   public static final String TRANSFORMER_REFERENCE_ELEMENT = "transformer";
   public static final String ANNOTATION_ELEMENT = "annotations";
-  public static final String PROCESSING_STRATEGY_ATTRIBUTE = "processingStrategy";
-  public static final String PROCESSING_STRATEGY_FACTORY_ATTRIBUTE = "processingStrategyFactory";
+  public static final String FILTER_ELEMENT_SUFFIX = "-filter";
```
Comment (dfeist) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ApplicationModel.java]: Is this part of this PR?

File: core-tests/src/test/java/org/mule/runtime/core/construct/FlowProcessingStrategyTestCase.java [hunk 5]
```diff
@@ -94,6 +102,37 @@ public void createDefaultProcessingStrategyIfNoneSpecified() throws Exception {
...
+                           TestProcessingStrategyFactory.class.getName(), () -> {
+                             MuleConfiguration muleConfiguration = new DefaultMuleConfiguration();
+                             when(muleContext.getConfiguration()).thenReturn(muleConfiguration);
+                             createFlow(null);
+                             assertEquals(flow.getProcessingStrategy().getClass(), TestProcessingStrategy.class);
+                           });
+  }
+
+
+  public static class TestProcessingStrategyFactory implements ProcessingStrategyFactory {
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/construct/FlowProcessingStrategyTestCase.java]: If you extend `AbstractProcessingStrategy` or an existing implementation then this would be bit more concise.

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ApplicationModel.java (hunk 4 above)
Comment (lucianoRM) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ApplicationModel.java]: Those constants are not used anymore

Comment (dfeist) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/model/ApplicationModel.java]: The filter constant, seems it was added.

File: core-tests/src/test/java/org/mule/runtime/core/construct/FlowProcessingStrategyTestCase.java [hunk 6]
```diff
@@ -94,6 +102,37 @@ public void createDefaultProcessingStrategyIfNoneSpecified() throws Exception {
...
+    @Override
+    public ProcessingStrategy create(MuleContext muleContext, String schedulersNamePrefix) {
+      return new TestProcessingStrategy();
+    }
+  }
+
+  private static class TestProcessingStrategy extends AbstractProcessingStrategy {
+
+    @Override
+    public Sink createSink(FlowConstruct flowConstruct, ReactiveProcessor pipeline) {
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/construct/FlowProcessingStrategyTestCase.java]: This is redundant.

File: core-tests/src/test/java/org/mule/runtime/core/internal/construct/DefaultFlowBuilderTestCase.java [hunk 7]
```diff
@@ -59,12 +61,12 @@ public void setUp() throws Exception {
     when(muleContext.getConfiguration()).thenReturn(muleConfiguration);
     when(muleConfiguration.getDefaultProcessingStrategyFactory()).thenReturn(defaultProcessingStrategyFactory);
     when(defaultProcessingStrategyFactory.create(any(), any())).thenReturn(processingStrategy);
+    when(muleContext.getSchedulerBaseConfig()).thenReturn(SchedulerConfig.config());
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/internal/construct/DefaultFlowBuilderTestCase.java]: Is this change needed, nothing changed re: scheduling did it?
